### PR TITLE
Introduce annotations

### PR DESCRIPTION
**Pull request by wafemand**
The new `Annotation` class replaces the `BlastHit` class in visualization. 

To manage annotations you can double click to annotation in the annotations widget. It opens a dialog window with checkboxes where you can select a view.

## No annotations selected
image

## Solid blast hit annotations selected
image

## Rainbow blast hit annotations selected
image

## Text selected
image

### CODE REVIEW

File: command_line/commoncommandlinefunctions.cpp [hunk 1]
```diff
@@ -1048,10 +1046,6 @@ NodeColourScheme getColourSchemeOption(QString option, QStringList * arguments)
         return UNIFORM_COLOURS;
     else if (colourString == "depth")
         return DEPTH_COLOUR;
-    else if (colourString == "blastsolid")
```
Comment (asl) [inline command_line/commoncommandlinefunctions.cpp:1051]: How we'd support annotations via cmdline now?

File: graph/annotationsmanager.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#ifndef BANDAGE_UI_ANNOTATIONSMANAGER_HPP_
+#define BANDAGE_UI_ANNOTATIONSMANAGER_HPP_
+
+#include <QObject>
+#include "graph/annotation.hpp"
+
+
+struct AnnotationGroup {
+    using AnnotationVector = std::vector<std::unique_ptr<Annotation>>;
+    using AnnotationMap = std::map<const DeBruijnNode *, AnnotationVector>;
```
Comment (asl) [inline graph/annotationsmanager.hpp]:
```suggestion
    using AnnotationMap = std::unordered_map<const DeBruijnNode *, AnnotationVector>;
```

File: graph/annotationsmanager.hpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+class AnnotationsManager : public QObject {
+Q_OBJECT
+
+public:
+    explicit AnnotationsManager(QObject *parent = nullptr) : QObject(parent) {}
+
+    //todo: replace with map???
+    using AnnotationGroupVector = std::vector<std::shared_ptr<AnnotationGroup>>;
```
Comment (asl) [inline graph/annotationsmanager.hpp]: why it is shared_ptr here? Can't annotation manager own the data?

File: graph/assemblygraph.h [hunk 4]
```diff
@@ -31,6 +31,7 @@
 #include "ogdf/basic/Graph.h"
 #include "ogdf/basic/GraphAttributes.h"
 #include "blast/blasthit.h"
+#include "annotation.hpp"
```
Comment (asl) [inline graph/assemblygraph.h]: do we need this include here?

File: program/globals.cpp [hunk 5]
```diff
@@ -33,6 +33,7 @@ double g_absoluteZoom;
 QSharedPointer<BlastSearch> g_blastSearch;
 QString m_tempDirectory;
 QSharedPointer<AssemblyGraph> g_assemblyGraph;
+std::shared_ptr<AnnotationsManager> g_annotationsManager;
```
Comment (asl) [inline program/globals.cpp:33]: Do we really need to have it shared_ptr here?

Comment (wafemand) [reply program/globals.cpp:33]: unique_ptr need a complete type here. I can't understand why...

File: command_line/commoncommandlinefunctions.cpp (hunk 1 above)
Comment (wafemand) [reply command_line/commoncommandlinefunctions.cpp:1051]: For now I saved backward compatibility for "--colour blastsolid", "--colour blastrainbow" and "--blasthits" in CLI. In future we need to implement a common way to set annotation setting from CLI.

**Pull request merged**